crates/core/tedge/src/cli/mapping/cli.rs

```diff
@@ -0,0 +1,126 @@
+use crate::cli::mapping::list::ListCommand;
+use crate::cli::mapping::test::TestCommand;
+use crate::command::BuildCommand;
+use crate::command::Command;
+use crate::ConfigError;
+use anyhow::anyhow;
+use anyhow::Context;
+use anyhow::Error;
+use std::path::PathBuf;
+use tedge_config::TEdgeConfig;
+use tedge_gen_mapper::pipeline::Message;
+use tedge_gen_mapper::MessageProcessor;
+
+#[derive(clap::Subcommand, Debug)]
+pub enum TEdgeMappingCli {
+    /// List pipelines and filters
+    List {
+        /// Path to pipeline and filter specs
+        ///
+        /// Default to /etc/tedge/gen-mapper
+        #[clap(long)]
+        mapping_dir: Option<PathBuf>,
+
+        /// List pipelines processing messages published on this topic
+        ///
+        /// If none is provided, lists all the pipelines
+        #[clap(long)]
+        topic: Option<String>,
+    },
+
+    /// Process message samples
+    Test {
+        /// Path to pipeline and filter specs
+        ///
+        /// Default to /etc/tedge/gen-mapper
+        #[clap(long)]
+        mapping_dir: Option<PathBuf>,
+
+        /// Path to the javascript filter or TOML pipeline definition
+        ///
+        /// If none is provided, applies all the matching pipelines
+        #[clap(long)]
+        filter: Option<PathBuf>,
+
+        /// Send a tick after all the message samples
+        #[clap(long = "final-tick")]
+        final_tick: bool,
+
+        /// Topic of the message sample
+        ///
+        /// If none is provided, messages are read from stdin expecting a line per message:
+        /// [topic] payload
```

Comment:
This nomenclature can be slightly confusing as on the command line, `[xxx]` generally means "optional", where as the topic is mandatory.

Though I assume `[]` was chosen because that is how we format the messages on the console from `tedge mqtt sub`. So given that, maybe we can just provide an example in addition to the syntax,

e.g.

```
/// [topic] payload
/// For example,
/// [te/example/one] {"text":"test message"}
```

Reply:
Yes, the bracketed topic syntax has been chosen to be able to pipe `tedge mqtt sub` output to `tedge mapping test`.

However, if the topic is provided on the command line then no brackets are expected:

```
$ tedge mapping test te/example/one `{"text":"test message"}`
```